bindata/network/openshift-sdn/sdn-ovs.yaml

```diff
@@ -35,16 +35,39 @@ spec:
           #!/bin/bash
           set -euo pipefail
           export SYSTEMD_IGNORE_CHROOT=yes
+          # systemctl may not be used in a separate PID namespace to reach
+          # the systemd running in PID 1. Therefore we need to use the dbus API
+          systemctl_is_active(){
+            gdbus call \
+              --system \
+              --dest org.freedesktop.systemd1 \
+              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
+              --method org.freedesktop.DBus.Properties.Get "org.freedesktop.systemd1.Unit" "ActiveState"  |
+                fgrep "(<'active'>,)" -q
+          }
+          systemctl_restart(){
+            gdbus call \
+              --system \
+              --dest org.freedesktop.systemd1 \
+              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
+              --method org.freedesktop.systemd1.Unit.Restart "replace"
+          }
+          svc_encode_name(){
+            # systemd encodes some characters, so far we only need to encode
+            # the character "-" but there may be more in the future.
+            echo "${1}" | sed 's/-/_2d/g'
```

Comment:
bash can do simple replaces itself using parameter expansion
```bash
echo "${1//-/_2d}"
```

      ${parameter/pattern/string}
              Pattern substitution.  The pattern is expanded to produce a pattern just as in pathname expansion, Parameter is expanded and the longest match of pattern against its value is replaced with string.  The match is performed using the rules described under Pattern Matching below.  If pattern begins
              with  /,  all matches of pattern are replaced with string.